public/app/features/alerting/unified/components/receivers/form/ChannelOptions.tsx

```diff
@@ -29,14 +29,15 @@ export function ChannelOptions<R extends ChannelValues>({
   readOnly = false,
 }: Props<R>): JSX.Element {
   const { watch } = useFormContext<ReceiverFormValues<R>>();
-  const currentFormValues = watch() as Record<string, any>; // react hook form types ARE LYING!
+  const currentFormValues = watch(); // react hook form types ARE LYING!
```

Comment:
🤔 Can Alerting team explain why we think the react hook types are wrong? Would like to help sort this out if we can 😌

Reply:
From my experience `watch` doesn't work well with recursive/nested types. Upgrading to the latest version of `react-hook-form` could fix those, but it's quite a big upgrade.

Reply:
Will this PR be merged? I want to add a new alert channel that depends on this PR?

Reply:
@aimuz can you please file an issue (or link to it if one exists) discussing adding another alerting channel? We need to discuss whether it makes sense to take on maintenance responsibilities for such a channel.

Reply:
@joeblubaugh  I created a question https://github.com/grafana/grafana/issues/55881

Reply:
Sorry, maybe I wasn't clear. You want to add a new type of notifier in `pkg/services/ngalert/notifier/channels`, right? Please update that issue to explain what the notifier would be. Is it another messaging service like Telegram or LINE? SMS? The maintainers need to understand what we would be responsible for after such a change was merged.

Reply:
Please take a look https://github.com/grafana/grafana/discussions/55883
 
 Also will the current PR not be merged? Or are you trying to fix this in some other way?

Reply:
Thanks for the update - I didn't want you to do a lot of work that wouldn't necessarily be accepted. I'll comment on that issue regarding WeCom-specific information.

I don't know about the current PR. I'll pull someone in. @gillesdemey PTAL

Reply:
These changes look 🆗 – we don't use the `showWhen` property for any receiver currently but I've verified that these changes do make it work as intended.

As for the types, like @Clarity-89 mentioned I believe this is because `react-hook-form` doesn't do well with nested forms and gets confused about the typings.